Source/carthage/Formatting.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Formatting.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 1/29/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import Commandant
+import Foundation
+import LlamaKit
+import PrettyColors
+import ReactiveCocoa
+
+extension Color.Wrap {
+	func autowrap(string: String) -> String {
+		return ColorOptions.colorful ? self.wrap(string) : string
+	}
+}
+
+internal struct Formatting {
+
+	static let bulletin = Color.Wrap(foreground: .Blue, style: .Bold)
+	static let bullets: String = {
+		return bulletin.autowrap("***") + " "
+	}()
+	
+	static let URL = [StyleParameter.Underlined] as Color.Wrap
+	static let projectName = [StyleParameter.Bold] as Color.Wrap
+	static let path = Color.Wrap(foreground: .Yellow)
+	
+	static func quote(string: String, quotationMark: String = "\u{0022}" /* double quote */) -> String {
+		return Color.Wrap(foreground: .Green).autowrap(quotationMark + string + quotationMark)
+	}
+	
+}
+
+internal struct Terminal {
+	static let term: String? = getEnvironmentVariable("TERM").value()
+	static let isDumb: Bool = (Terminal.term?.lowercaseString as NSString?)?.isEqualToString("dumb") ?? false
+	static let isTTY: Bool = isatty(STDOUT_FILENO) == 1
+}
+
+public enum ColorArgument: String, ArgumentType {
+	case Auto = "auto"
+	case Never = "never"
+	case Always = "always"
+	
+	public var isColorful: Bool {
```

Comment:
I named the `isX` variables following the example of the `stdlib`’s `Array.isEmpty` and the global `isEmpty`.

Doing a bit more poking around, [`Float`](http://swiftdoc.org/type/Float/) and [`Double`](http://swiftdoc.org/type/Double/) have tons of `isX` variables. Even a few `hasX` variables (<a href='http://swiftdoc.org/type/String/' title='hasPrefix, hasSuffix'>hasPrefix, hasSuffix</a>, <a href='http://swiftdoc.org/type/StaticString/' title='hasPointerRepresentation'>hasPointerRepresentation</a>).

But, if the GitHub style is to avoid `isX`, then I'm happy to comply.